lib/cerner/oauth1a/access_token_agent.rb

```diff
@@ -215,6 +238,19 @@ def convert_to_http_uri(access_token_url)
         uri
       end
 
+      # Internal: Returns a String containing the canonical root url.
+      #
+      # url - A URL to get the canonical root url String from.
+      #
+      # raises ArgumentError if url is nil.
+      def canonical_root_url_for(url)
+        raise ArgumentError, 'url is nil' unless url
+
+        realm = "#{url.scheme}://#{url.host}"
+        realm += ":#{url.port}" unless url.port == 80 || url.port == 443
```

Comment:
Could this use 'url.default_port', instead of hard coding the check for 80 and 443? default_port should return 80 if this is an HTTP URI and 443 is if this an HTTPS URI.

Alternatively, we could create a new URI and to_s it. Maybe like this:
```
realm = URI.new(url.scheme, nil, url.host, url.port)
realm.to_s
```

Reply:
I'm not finding that issue with all of the versions of Ruby we need to support:

```
2.6.0 :004 > URI('http://localhost/test').to_s
 => "http://localhost/test" 

2.5.3 :001 > URI('http://localhost/test').to_s
 => "http://localhost/test"

2.4.5 :001 > URI('http://localhost/test').to_s
 => "http://localhost/test"
```

But I think I see the issue you're talking about. 
```
2.6.0 :005 > URI::Generic.new('https', nil, 'localhost', 80, nil, nil, nil, nil, nil).to_s
 => "https://localhost:80"
```
When you use URI::Generic, it does put the port in it. What we need is to construct a URI::HTTP or a URI:HTTPS.

Reply:
I'm still thinking about this, but here's an alternative ...

```
2.6.0 :008 > url = URI("https://cerner.com/oauth/access")
 => #<URI::HTTPS https://cerner.com/oauth/access> 
2.6.0 :009 > realm = URI("#{url.scheme}://#{url.host}:#{url.port}")
 => #<URI::HTTPS https://cerner.com> 
```

Reply:
This got me thinking, and technically the assumption of the realm being the canonical root URI of the access token URL is only valid for Cerner's implementation of OAuth1a. Based on the README of this library it reads like it's only intended for use with Cerner's OAuth1a and not with other random OAuth1a protected services, so this assumption should be fine.

Reply:
Switched to use 'URI' to extract the realm: https://github.com/cerner/cerner-oauth1a/pull/4/commits/ac3f7b142d01f69cdc2de378b3120538f42a54f4